File: tests/fixtures/realistic/csharp_multihop_broken/QueryHelper.cs
// Sink helper (fileC) for the csharp_multihop fixture.
//
// RunQuery() concatenates its parameter into a SQL string and passes it to a
// SqlCommand sink. Its single-file summary records params_to_sink for param 0
// with rule csharp/taint-sql-injection. Scanned alone, `term` is just a
// parameter (not a source), so no taint finding fires here.

using System.Data.SqlClient;

class QueryHelper {
    public static void RunQuery(string term) {
        string sql = "SELECT * FROM users WHERE name = '" + term + "'";
        var cmd = new SqlCommand(sql);
        cmd.ExecuteReader();
    }
}

File: tests/fixtures/realistic/java_multihop_broken/QueryHelper.java
// Sink helper (fileC) for the java_multihop_broken fixture.
//
// Same sink as the positive fixture. The chain is broken upstream (in
// Service.process), so the only finding here is the single-file
// java/no-sql-injection regex heuristic on the concatenation.

import java.sql.Statement;

class QueryHelper {
    static Statement stmt;

    static void runQuery(String term) throws Exception {
        stmt.executeQuery("SELECT * FROM users WHERE name = '" + term + "'");
    }
}
